Sources/PostgresNIO/Connection/PostgresConnection+Notifications.swift

```diff
@@ -0,0 +1,56 @@
+import NIO
+
+public final class PostgresListenContext {
+    var stopper: (() -> Void)?
+
+    public func stop() {
+        stopper?()
+        stopper = nil
+    }
+}
+
+extension PostgresConnection {
+    @discardableResult
+    public func listen(channel: String, handler notificationHandler: @escaping (PostgresMessage.NotificationResponse, PostgresListenContext) -> Void) -> PostgresListenContext {
+        let listenContext = PostgresListenContext()
+        let channelHandler = PostgresNotificationHandler(channel: channel, notificationHandler: notificationHandler, listenContext: listenContext)
+        let pipeline = self.channel.pipeline
+        _ = pipeline.addHandler(channelHandler, name: nil, position: .before(requestHandler))
```

Comment:
Since `PostgresRequestHandler` is ignoring `.notificationResponse` now, I think you should be able to simply append the `PostgresNotificationHandler`. That would make thing simpler not needing to pass around `requestHandler`.

Reply:
Nit: Please use `self.` whenever you access a property (pretend you're in an escaping closure all the time).